File: Src/java/cql-to-elm/src/jvmTest/java/org/cqframework/cql/cql2elm/TestFhirModelInfoProvider.java
package org.cqframework.cql.cql2elm;

import static kotlinx.io.CoreKt.buffered;
import static kotlinx.io.JvmCoreKt.asSource;
import static org.hl7.elm_modelinfo.r1.serializing.XmlModelInfoReaderKt.parseModelInfoXml;

import org.hl7.cql.model.ModelIdentifier;
import org.hl7.cql.model.ModelInfoProvider;
import org.hl7.elm_modelinfo.r1.ModelInfo;

/**
 * Created by Bryn on 12/11/2016.
 */
public class TestFhirModelInfoProvider implements ModelInfoProvider {
    private Class clazz;

    public TestFhirModelInfoProvider(Class clazz) {
        this.clazz = clazz;
    }

    public ModelInfo load(ModelIdentifier modelIdentifier) {
        if (modelIdentifier.getId().equals("FHIR")) {
            var source = buffered(asSource(clazz.getResourceAsStream("fhir-modelinfo-1.8.xml")));
            return parseModelInfoXml(source);
        }

        return null;
    }
}

File: cql-to-elm/src/jvmTest/kotlin/org/cqframework/cql/cql2elm/TestFhirModelInfoProvider.kt
package org.cqframework.cql.cql2elm

import kotlinx.io.asSource
import kotlinx.io.buffered
import org.hl7.cql.model.ModelIdentifier
import org.hl7.cql.model.ModelInfoProvider
import org.hl7.elm_modelinfo.r1.ModelInfo
import org.hl7.elm_modelinfo.r1.serializing.parseModelInfoXml

/** Created by Bryn on 12/11/2016. */
class TestFhirModelInfoProvider(private val clazz: Class<*>) : ModelInfoProvider {
    override fun load(modelIdentifier: ModelIdentifier): ModelInfo? {
        if (modelIdentifier.id == "FHIR") {
            val source = clazz.getResourceAsStream("fhir-modelinfo-1.8.xml")!!.asSource().buffered()
            return parseModelInfoXml(source)
        }

        return null
    }
}
